### PR TITLE
Fixing Integration Problems

### LINKED ISSUE #12: Fix Branch Integration Problems

### PR DESCRIPTION
**Pull request by FrankKleiton**
fixes #12 

Captura de tela_2020-07-29_15-34-59

A falta do exact e a existência de um arquivos routes no mesmo nível de diretório da pasta routes, junto com a falta de um exact gerou alguns problemas para carregar os componentes da aplicação. Também havia alguns erros que cometi como usar async no use effect, e a falta de um pouco mais de tratamento no frontend para alguns casos especiais. (Arrays vazios nos materials e loans)

### CODE REVIEW

File: src/pages/Material/index.js [hunk 1]
```diff
@@ -11,61 +11,61 @@ const Material = () => {
...
+                }
+              );
+            } else {
+              l.loan_time = formatRelative(
+                toDate(parseISO(l.loan_time)),
+                new Date(),
+                {
+                  locale: pt,
+                }
+              );
```
Comment (hdev14) [inline src/pages/Material/index.js:41]: Well, when we are using async into useEffect, we can't use async before callback. To do that, we have to create an async function inside of useEffect, then we call it normaly without await. For instance:
```js
useEffect(() => {
  async function fetchData() { /* piece of code */ }
  fetchData();
}, []);
```

**Review by hdev14 — COMMENTED** [review]
The history package were with a issue that after a redirect the page doesn't show. So check if the version is equal 4.13 (something like that).

**Comment by FrankKleiton** [comment]
> The history package were with a issue that after a redirect the page doesn't show. So check if the version is equal 4.13 (something like that).

Indeed that was happening, but it was because you forgot the exact keyword in the route, yerterday I didn't spended much time debuging so i didn't saw, but today I perceived when putted some console.log(). The worst part in coding the frontend is that debuging is very exaustive and boring so that kind of thing most the time escapes our sight, even when we are coding and not debuging in the moment.

File: src/pages/Material/index.js (hunk 1 above)
Comment (FrankKleiton) [reply src/pages/Material/index.js:41]: > Well, when we are using async into useEffect, we can't use async before callback. To do that, we have to create an async function inside of useEffect, then we call it normaly without await. For instance:
> 
> ```js
> useEffect(() => {
>   async function fetchData() { /* piece of code */ }
>   fetchData();
> }, []);
> ```

Yeah, I got that when I was coding, but when returning the variable with the await result from the function, I actually get it an promise pending, so I would need to resolve it again, so I make in the old way and put a then there lol

**Comment by hdev14** [comment]
Yeah, it's horrible to deal with debugging in frontend. I usually use React Dev Tool to see most of the variables, but in most of cases only a console.log is helpful.

**Pull request merged**